src/main/kotlin/com/hibob/academy/service/OwnerService.kt

```diff
@@ -0,0 +1,22 @@
+package com.hibob.academy.service
+
+import com.hibob.academy.dao.OwnerCreationRequest
+import com.hibob.academy.dao.OwnerDao
+import com.hibob.academy.dao.OwnerData
+import jakarta.inject.Inject
+import org.springframework.stereotype.Service
+
+@Service
+class OwnerService @Inject constructor(private val ownerDao: OwnerDao) {
+
+    fun createOwner(owner: OwnerCreationRequest) = ownerDao.createOwner(owner)
+
+    fun getAllOwners(companyId: Long) = ownerDao.getAllOwners(companyId)
+
+    fun getOwnerByPetId(petId: Long, companyId: Long): OwnerData? {
+        val ownerInfo = ownerDao.getOwnerByPetId(petId, companyId)
+        ownerInfo?.let{
+            return ownerInfo
+        } ?: throw IllegalArgumentException("Pet doesn't have an owner")
```

Comment:
Is it an error no to have owner? You return OwnerData? - so it seems you allow nullable

Reply:
it's not an error but if im trying to get a pets owner and it doesnt have one so it's an error

Reply:
But it is nullable -it is allowed that a pet will have no owner 
Not sure I agree with throwing error here- but it not a functional bug so if you think this is correct, keep it

Reply:
The question is what are my options instead of throwing an exception. if the pet has no owner then just return null maybe?